### PR TITLE
fix: macOS libzim <-> icu4c again and make mismatch a fatal error

### PR DESCRIPTION
**Pull request by shenlebantongying**
Homebrew is in a long migration of icu 75->76.

Now, libzim depends on icu4c@76.

Add rudimentary code to verify it.

(Random: 

I don't like any of these, but let's work with this for now.

I tried to use vcpkg on macOS before, but it doesn't work starting from here -> https://github.com/microsoft/vcpkg/pull/39600)

### CODE REVIEW

File: cmake/Deps_Unix.cmake [hunk 1]
```diff
@@ -82,13 +77,31 @@ if (WITH_EPWING_SUPPORT)
...
+        message("Zim include dirs -> ${ZIM_INCLUDE_DIRS}")
+        message("Homebrew icu include dirs-> ${BREW_ICU_INCLUDE_DIRS}")
+
+        list(GET BREW_ICU_INCLUDE_DIRS 0 ONE_OF_BREW_ICU_INCLUDE_DIR)
+        if (ONE_OF_BREW_ICU_INCLUDE_DIR IN_LIST ZIM_INCLUDE_DIRS)
+            message("ZIM OK!")
+        else ()
+            message(FATAL_ERROR "!!!! ZIM <-> icu error -> check `brew info libzim` and `brew list libzim`")
+        endif ()
+        # TODO: get rid of these 💩, how?
```
Comment (xiaoyifang) [inline cmake/Deps_Unix.cmake:104]: > 💩,

not proper  here.

Comment (shenlebantongying) [reply cmake/Deps_Unix.cmake:104]: ok :)

**Comment by xiaoyifang** [comment]
is this a libzim dependence issue?  If yes, can report this to libzim project.

**Comment by xiaoyifang** [comment]
https://cmake.org/cmake/help/latest/module/FindICU.html

**Comment by xiaoyifang** [comment]
> Windows build too, but it doesn't work starting from here -> [microsoft/vcpkg#39600](https://github.com/microsoft/vcpkg/pull/39600))

the link is opencc?

**Comment by shenlebantongying** [comment]
FindICU finds the icu provided by Apple's SDK (? or system itself)

We need the one from homebrew which libzim linked with.

**Comment by xiaoyifang** [comment]
Is this some libzim link problem?

**Comment by xiaoyifang** [comment]
https://github.com/xiaoyifang/goldendict-ng/actions/runs/11875181316
Do not work now.

**Comment by shenlebantongying** [comment]
fixed by https://github.com/xiaoyifang/goldendict-ng/pull/1960

**Pull request merged**